lua/lean/abbreviations.lua

```diff
@@ -33,18 +37,150 @@ local function snippets_nvim_enable(snippets, lean_abbreviations)
   snippets.snippets = all_snippets
 end
 
+local abbr_mark_ns = vim.api.nvim_create_namespace('leanAbbreviationMark')
+
+local function get_extmark_range(abbr_ns, id, buffer)
+  local row, col, details = unpack(
+    vim.api.nvim_buf_get_extmark_by_id(buffer or 0, abbr_ns, id, {details = true}))
+  return row, col, details and details.end_row, details and details.end_col
+end
+
+function M._clear_abbr_mark()
```

Comment:
All these I think should be private (i.e. not exposed on the module) I think (though I'll be damned if I claimed I understood Lua scoping norms yet) -- but I think:

```suggestion
local function _clear_abbr_mark()
```

besides for the one you need to put on `_G`?

Reply:
I've made all functions local that are not called from vimscript.  (For some reason I thought that if a function needs access to `M` it needs to be called `M.frob`.)